src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -837,56 +842,181 @@ def create_dns_record_set(resource_group_name, zone_name, record_set_name, recor
+        'ptr_records': 'ptr',
+        'soa_record': 'soa',
+        'srv_records': 'srv',
+        'txt_records': 'txt'
+        }
+
+    zone_obj = {
+        '$origin': zone_name.rstrip('.') + '.'
+        }
+
+    for record_set in record_sets:
+        for property_name, record_type in record_property_types.items():
+            record_data = getattr(record_set, property_name, None)
+            if not record_data:
+                continue
+            if not isinstance(record_data, list):
+                record_data = [record_data]
+            if record_type not in zone_obj:
+                zone_obj[record_type] = []
+            for record in record_data:
+                record_obj = {'ip': record.ipv6_address} if record_type == 'aaaa' \
+                    else {'ip': record.ipv4_address} if record_type == 'a' \
+                    else {'alias': record.cname} if record_type == 'cname' \
+                    else {'preference': record.preference, 'host': record.exchange} \
+                    if record_type == 'mx' \
+                    else {'host': record.nsdname} if record_type == 'ns' \
+                    else {'host': record.ptrdname} if record_type == 'ptr' \
+                    else {'mname': record.host, 'rname': record.email,
+                          'serial': record.serial_number, 'refresh': record.refresh_time,
+                          'retry': record.retry_time, 'expire': record.expire_time,
+                          'minimum': record.minimum_ttl} if record_type == 'soa' \
+                    else {'priority': record.priority, 'weight': record.weight,
+                          'port': record.port, 'target': record.target} if record_type == 'srv' \
+                    else {'txt': ' '.join(record.value)} if record_type == 'txt' \
+                    else None
+                record_obj['name'] = record_set.name
+                record_obj['ttl'] = record_set.ttl
+                zone_obj[record_type].append(record_obj)
+
+    if 'soa' in zone_obj:
+        zone_obj['soa'] = zone_obj['soa'][0]
+        zone_obj['soa']['mname'] = zone_obj['soa']['mname'].rstrip('.') + '.'
+        zone_obj['soa']['rname'] = zone_obj['soa']['rname'].rstrip('.') + '.'
+
+    zone_file_text = make_zone_file(zone_obj)
+
+    with open(file_name, 'w') as f:
+        f.write(zone_file_text)
+
+def import_zone(resource_group_name, zone_name, file_name, location='global'):
+    file_text = None
+    with open(file_name) as f:
+        file_text = f.read()
+    zone_obj = parse_zone_file(file_text)
+
+    zone_origin = zone_obj['$origin'].rstrip('.')
+    if zone_name != zone_origin:
+        raise CLIError('Zone file origin "{}" does not match zone name "{}"'
+                       .format(zone_origin, zone_name))
+
+    client = get_mgmt_service_client(DnsManagementClient)
+
+    try:
+        if client.zones.get(resource_group_name, zone_name):
+            raise CLIError('Zone "{}" already exists'.format(zone_name))
```

Comment:
Since this is a behavior change from xplat, I suggest add more text say "please delete the existing and re-import"?